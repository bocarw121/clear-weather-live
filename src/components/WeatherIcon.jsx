import React from "react";
import { StyleSheet } from "react-native";

import { iconUrl } from "../api/urls";
import Loading from "./Loading";
import Image from "react-native-remote-svg";

const WeatherIcon = ({ icon, id }) => {
  if (!icon) return <Loading size="Large" />;

  return (
    <Image
      source={{
        uri: iconUrl(icon, id),
      }}
      style={styles.image}
    />
  );
};

export default WeatherIcon;

const styles = StyleSheet.create({
  image: { width: 150, height: 150, position: "relative", top: 5 },
});
