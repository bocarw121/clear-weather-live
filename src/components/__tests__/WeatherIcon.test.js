import React from "react";

import { render } from "@testing-library/react-native";

import WeatherIcon from "../WeatherIcon";

it("Should render the WeatherIcon", () => {
  const screen = render(<WeatherIcon />);

  expect(screen).toMatchSnapshot();
});
