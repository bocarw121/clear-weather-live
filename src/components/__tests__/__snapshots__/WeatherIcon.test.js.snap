// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the WeatherIcon 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": "80%",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="Large"
  />
</View>
`;
